src/main/ArrayVisualizer.java

```diff
@@ -494,6 +498,10 @@ public void run() {
         this.drawWindows();
     }
 
+    public static ArrayVisualizer getInstance() {
```

Comment:
it seems a bit hacky to have a global reference to the current instance instead of like passing the instance to whatever constructor it's needed in, but maybe not

Reply:
This is a common pattern with singletons (and kinda the whole point of them). Maybe ArrayVisualizer should throw an exception if you try to create another instance.

Reply:
how do other ArrayV classes get their reference to `ArrayVisualizer`?

Reply:
Through references passed all over the place. However, this doesn't exactly work for a class that gets instantiated outside of internal code. What I *could* do is assign some public field somewhere, but that would be less clean.

Reply:
you could just pass the `ArrayVisualizer` reference to the constructor though, right?

Reply:
That would be inconvenient for sorts, besides being inconsistent with `ArrayList`.

Reply:
it's already inconsistent by its methods' additional arguments (which are optional but just because they have sensible defaults), and anyway, imo it makes sense semantically because, unlike regular lists, `ArrayVList` interacts with the `ArrayVisualizer`

Reply:
It would still be inconvenient, harder to convert, and it's inconsistent compared to other Java programs.

Reply:
why'd it be inconvenient?

Reply:
You'd have to write `new ArrayVList(arrayVisualizer)` instead of just `new ArrayVList();`.

Reply:
ah should've clarified, I mean that you already have to use those tricky `Reads` and `Writes` methods, isn't this a pretty minor thing?

Reply:
Yes, but there's a general rule that you shouldn't  need to use `arrayVisualizer` inside a sort.

Reply:
why?

Reply:
Because it's more low-level

Reply:
what about creating a `Writes` method for creating/deleting ArrayVLists? (it's indirect, but that's how aux arrays work)
and imo, passing `arrayVisualizer` is fine since it's what represents the array visualizer as a whole

Reply:
A creation method could be added... but the deletion one already exists

Reply:
https://github.com/Gaming32/ArrayV-v4.0/pull/66/commits/ed2c4044e19a3315827d6ce4659edb849e6474fc

Reply:
given that there's now a `Writes` creation method, why not pass `ArrayVisualizer` to the constructor there? unless you wanna transition to a static singleton instance kinda structure, in which alright

Reply:
I like the singleton system, as it's used somewhere else in the code now too.

Reply:
alright cool